lib/owl/data/sequence.ex

```diff
@@ -0,0 +1,102 @@
+defmodule Owl.Data.Sequence.Helper do
+  @moduledoc false
+
+  defmacro defsequence_type(name, type, define_name_by_sequence? \\ true) do
+    quote bind_quoted: [
+            name: name,
+            type: type,
+            define_name_by_sequence?: define_name_by_sequence?
+          ] do
+      if define_name_by_sequence? do
+        seq = apply(IO.ANSI, name, [])
+        defp name_by_sequence(unquote(seq)), do: unquote(name)
+      end
+
+      defp type_by_name(unquote(name)), do: unquote(type)
+    end
+  end
+end
+
+defmodule Owl.Data.Sequence do
+  @moduledoc false
+
+  import Owl.Data.Sequence.Helper
+
+  @doc """
+  Get the sequence name of an escape sequence or `nil` if not a sequence.
+  """
+  def ansi_to_name(binary) when is_binary(binary) do
+    name_by_sequence(binary)
+  end
+
+  @doc """
+  Get the sequence type of an escape sequence or `nil` if not a sequence.
+  """
+  def ansi_to_type(binary) when is_binary(binary) do
+    if name = ansi_to_name(binary) do
```

Comment:
We should support RGB colors as well

Reply:
I _believe_ this is already supported, but working on adding tests now. For instance, these two tests are currently passing. Is this what you mean?

```elixir
      assert to_from_ansidata(["Hello ", Owl.Data.tag("world", IO.ANSI.color(161)), "!"]) ==
               [["Hello ", Owl.Data.tag("world", "\e[38;5;161m")], "!"]

      assert to_from_ansidata([
               "Hello ",
               Owl.Data.tag("world", IO.ANSI.color_background(161)),
               "!"
             ]) == [["Hello ", Owl.Data.tag("world", "\e[48;5;161m")], "!"]
```